dask/array/tests/test_routines.py

```diff
@@ -1329,6 +1329,14 @@ def test_roll(chunks, shift, axis):
         assert_eq(np.roll(x, shift, axis), da.roll(a, shift, axis))
 
 
+def test_roll_bug():
+    # This bug was exposed in GitHub Issue #8723
+    x = da.arange(2, 3)
+    y = da.roll(x, 1)
+    y[0] = 0
+    assert x[0].compute() == 2
```

Comment:
It'd be simpler to assert what we're actually testing.

```suggestion
    assert y is not x
```